libs/aws/langchain_aws/chat_models/bedrock.py

```diff
@@ -507,14 +601,167 @@ def bind_tools(
+                    '''An answer to the user question along with justification for the answer.'''
+                    answer: str
+                    justification: str
+
+                llm =ChatBedrock(
+                    model_id="anthropic.claude-3-sonnet-20240229-v1:0",
+                    model_kwargs={"temperature": 0.001},
+                )  # type: ignore[call-arg]
+                structured_llm = llm.with_structured_output(AnswerWithJustification, include_raw=True)
+
+                structured_llm.invoke("What weighs more a pound of bricks or a pound of feathers")
+                # -> {
+                #     'raw': AIMessage(content='', additional_kwargs={'tool_calls': [{'id': 'call_Ao02pnFYXD6GN1yzc0uXPsvF', 'function': {'arguments': '{"answer":"They weigh the same.","justification":"Both a pound of bricks and a pound of feathers weigh one pound. The weight is the same, but the volume or density of the objects may differ."}', 'name': 'AnswerWithJustification'}, 'type': 'function'}]}),
+                #     'parsed': AnswerWithJustification(answer='They weigh the same.', justification='Both a pound of bricks and a pound of feathers weigh one pound. The weight is the same, but the volume or density of the objects may differ.'),
+                #     'parsing_error': None
+                # }
+
+        Example: Dict schema (include_raw=False):
+            .. code-block:: python
+
+                from langchain_aws.chat_models.bedrock import ChatBedrock
+
+                schema = {
+                    "name": "AnswerWithJustification",
+                    "description": "An answer to the user question along with justification for the answer.",
+                    "input_schema": {
+                        "type": "object",
+                        "properties": {
+                            "answer": {"type": "string"},
+                            "justification": {"type": "string"},
+                        },
+                        "required": ["answer", "justification"]
+                    }
+                }
+                llm =ChatBedrock(
+                    model_id="anthropic.claude-3-sonnet-20240229-v1:0",
+                    model_kwargs={"temperature": 0.001},
+                )  # type: ignore[call-arg]
+                structured_llm = llm.with_structured_output(schema)
+
+                structured_llm.invoke("What weighs more a pound of bricks or a pound of feathers")
+                # -> {
+                #     'answer': 'They weigh the same',
+                #     'justification': 'Both a pound of bricks and a pound of feathers weigh one pound. The weight is the same, but the volume and density of the two substances differ.'
+                # }
+
+        """  # noqa: E501
+        if "claude-3" not in self._get_model():
+            warnings.warn(
+                "Structured output is only supported for claude-3 models on Bedrock."
+            )
```

Comment:
should we return here? or even error? rest of the code won't work anyways right

Reply:
Yes we can error 👍